nan.h

```diff
@@ -801,44 +801,83 @@ class TryCatch {
 #endif
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::Function> func
     , int argc
     , v8::Local<v8::Value>* argv) {
+    v8::Isolate *isolate = v8::Isolate::GetCurrent();
+    v8::EscapableHandleScope scope(isolate);
+    v8::Local<v8::Object> target_obj;
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
-    return New(node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv));
+    if (target->IsUndefined() || target->IsNull()) {
+      target_obj = isolate->GetCurrentContext()->Global();
```

Comment:
I believe the `val_` pointer is, but internally the`Value` has a check to see what the type is. So it can determine that way.